jdaviz/configs/imviz/plugins/tools.py

```diff
@@ -68,3 +69,61 @@ def on_mouse_or_key_event(self, data):
                     viewer.state.x_max = x + dx / 2
                     viewer.state.y_min = y - dy / 2
                     viewer.state.y_max = y + dy / 2
+
+
+@viewer_tool
+class BqplotContrastBias(CheckableTool):
+
+    icon = 'glue_contrast'
+    tool_id = 'bqplot:contrastbias'
+    action_text = 'Adjust contrast/bias'
+    tool_tip = 'Click and drag to adjust, double-click to reset'
+
+    def __init__(self, viewer, **kwargs):
+        self._time_last = 0
+        super().__init__(viewer, **kwargs)
+
+    def activate(self):
+        self.viewer.add_event_callback(self.on_mouse_or_key_event,
+                                       events=['dragstart', 'dragmove',
+                                               'dragend', 'dblclick'])
+
+    def deactivate(self):
+        self.viewer.remove_event_callback(self.on_mouse_or_key_event)
+
+    def on_mouse_or_key_event(self, data):
+
+        event = data['event']
+
+        # Note that we throttle this to 200ms here as changing the contrast
+        # and bias it expensive since it forces the whole image to be redrawn
+        if event == 'dragmove':
+            if (time.time() - self._time_last) <= 0.2:
+                return
+
+            event_x = data['domain']['x']
+            event_y = data['domain']['y']
+
+            if ((event_x < self.viewer.state.x_min) or
```

Comment:
@duytnguyendtn , does it behave better for you with this check in?

@astrofrog should re-review the logic here as well.